### PR TITLE
Upgrade 'Voting' sample

### PR DESCRIPTION
**Pull request by IEvangelist**
Relates to #8035

This currently comments out the Redis bits, as the NuGet packages do not yet support Orleans 7.0.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8138)

### CODE REVIEW

File: samples/Voting/Data/DemoService.cs [hunk 1]
```diff
@@ -1,42 +1,37 @@
...
 
-        public async Task SimulateVoters(string pollId, int numVotes)
+    public async Task SimulateVoters(string pollId, int numVotes)
+    {
+        try
         {
-            try
+            var pollGrain = _grainFactory.GetGrain<IPollGrain>(pollId);
+            var results = await pollGrain.GetCurrentResults();
+            var random = new Random();
```
Comment (gfoidl) [inline samples/Voting/Data/DemoService.cs:22]: To show "good behavior" in the samples should this use `Random.Shared` instead?
I know that totally unrelated to Orleans, but to make people aware that shared-instance avoid common troubles that may arise when new Random is used wrongly (mostly for beginners).

File: samples/Voting/Grains/UserAgentGrain.cs [hunk 2]
```diff
@@ -1,109 +1,100 @@
...
-                throw new InvalidOperationException("You have already created 5 polls, which is enough for anybody.");
-            }
+        // Generate a new id and get a reference to the PollGrain with that id
+        var pollId = Guid.NewGuid().ToString("N")[..6];
+        var pollGrain = _grainFactory.GetGrain<IPollGrain>(pollId);
 
-            // Generate a new id and get a reference to the PollGrain with that id
-            var pollId = Guid.NewGuid().ToString("N").Substring(0, 6);
-            var pollGrain = _grainFactory.GetGrain<IPollGrain>(pollId);
+        // Create the poll. We could avoid colitions here by making this return an error if a poll
```
Comment (gfoidl) [inline samples/Voting/Grains/UserAgentGrain.cs:55]:
```suggestion
        // Create the poll. We could avoid collitions here by making this return an error if a poll
```
?

File: samples/Voting/Program.cs [hunk 3]
```diff
@@ -19,12 +14,12 @@
     else
     {
         // In Kubernetes, we use environment variables and the pod manifest
-        orleansBuilder.UseKubernetesHosting();
+        //orleansBuilder.UseKubernetesHosting();
```
Comment (gfoidl) [inline samples/Voting/Program.cs:17]: Should be uncommented for !Development?

File: samples/Voting/Data/DemoService.cs (hunk 1 above)
Comment (IEvangelist) [inline samples/Voting/Data/DemoService.cs:22]:
```suggestion
            var random = Random.Shared;
```

File: samples/Voting/Program.cs (hunk 3 above)
Comment (IEvangelist) [reply samples/Voting/Program.cs:17]: The underlying package dependencies need to be updated to 7.0. Can't right now, this should probably be a draft.

**Review by ReubenBond — APPROVED** [review]

**Pull request merged**